include/zephyr/posix/signal.h

```diff
@@ -12,6 +12,60 @@
 extern "C" {
 #endif
 
+#ifdef CONFIG_POSIX_SIGNAL
+#define _NSIG       64
+#define _NSIG_BPW   BITS_PER_LONG
+#define _NSIG_WORDS (_NSIG / _NSIG_BPW)
+
+#define SIGHUP    1
+#define SIGINT    2
+#define SIGQUIT   3
+#define SIGILL    4
+#define SIGTRAP   5
+#define SIGABRT   6
+#define SIGBUS    7
+#define SIGFPE    8
+#define SIGKILL   9
+#define SIGUSR1   10
+#define SIGSEGV   11
+#define SIGUSR2   12
+#define SIGPIPE   13
+#define SIGALRM   14
+#define SIGTERM   15
+/* 16 */
+#define SIGCHLD   17
+#define SIGCONT   18
+#define SIGSTOP   19
+#define SIGTSTP   20
+#define SIGTTIN   21
+#define SIGTTOU   22
+#define SIGURG    23
+#define SIGXCPU   24
+#define SIGXFSZ   25
+#define SIGVTALRM 26
+#define SIGPROF   27
+/* 28 */
+#define SIGPOLL   29
+/* 30 */
+#define SIGSYS    31
+#define SIGUNUSED 31
+
+#define SIGRTMIN 32
+#ifndef SIGRTMAX
+#define SIGRTMAX _NSIG
+#endif
+
+typedef struct {
+	unsigned long sig[_NSIG_WORDS];
+} sigset_t;
```

Comment:
we should disable this in `scripts/checkpatch.pl` for posix headers, it's a false positive.